docs/rules/consistent-this.md

```diff
@@ -47,6 +55,20 @@ This rule is disabled by default. You can configure the designated `this` variab
 "consistent-this": [0, "self"]
 ```
 
+You can also remove the requirement that a declaration of the `this` variable must be immediately assigned with the `allow-empty-var` option.
+
+```js
+"consistent-this": [0, "self", "allow-empty-var"]
+```
```

Comment:
We need to work on the options for this. I don't think `allow-empty-var` describes what happens very well. Also, these options are usually represented as booleans on an options object rather than a string.